nexus/src/authn/mod.rs

```diff
@@ -326,6 +319,22 @@ impl Actor {
             Actor::SiloUser { silo_id, .. } => Some(*silo_id),
         }
     }
+
+    pub fn silo_user_id(&self) -> Option<Uuid> {
+        match self {
+            Actor::UserBuiltin { .. } => None,
+            Actor::SiloUser { silo_user_id, .. } => Some(*silo_user_id),
+        }
+    }
+}
+
+impl From<&Actor> for db::model::IdentityType {
```

Comment:
Just out of curiosity, why the change from `actor_type()` to a `From` impl?  It seems the same signature (`&Actor -> db::model::IdentityType`) and implementation.  (I know I'd suggested that we use a separate return type for `actor_type()` that would be an enum of just `UserBuiltin` and `SiloUser` (but not `SiloGroup`).  This change doesn't change that, which is fine.  I was wondering if there was some other reason you preferred `From` to a function.)

Reply:
In situations where a section of code has an `Actor`, I think this reduces the possibility that code can ask the wrong question with `match`, like if the Actor is somehow a SiloGroup. Previously, the code matched on the result of a function call:

```rust
match actor.actor_type() {
    SiloGroup => { ... }
}
```

Here it seems like a natural question to see if an actor is a group, which doesn't make sense because the Actor enum doesn't contain that variant.

With the From, the user would have to type something like

```rust
let actor_type: db::model::IdentityType = actor.into();
match actor_type {
    SiloGroup => { ... }
}
```

or

```rust
match actor.into::<db::model::IdentityType>() {
    SiloGroup => { ... }
}
```

Which is explicitly asking questions about the db identity type, not what actor is - it's much more natural to type `match actor {` instead.